### PR TITLE
Add Synthetix PerpsV2 to perpetual_trades abstraction

### PR DESCRIPTION
**Pull request by RplusT**
Brief comments on the purpose of your changes:
This PR adds the perpetual trades related to V2 of Synthetix' perpetuals trading (Synthetix PerpsV2). It launched on December 2022 and enabled 22 new asset markets earlier this month (February 2023). Naming conventions have also been modified to accommodate this new version.

**For Dune Engine V2**

I've checked that:

### General checks:
* [x] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
* [x] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
* [x] if adding a new model, I added a test
* [x] the filename is unique and ends with .sql
* [x] each sql file is a select statement and has only one view, table or function defined  
* [x] column names are `lowercase_snake_cased`
* [ ] if adding a new model, I edited the dbt project YAML file with new directory path for both models and seeds (if applicable)
* [x] if wanting to expose a model in the UI (Dune data explorer), I added a post-hook in the JINJA config to add metadata (blockchains, sector/project, name and contributor Dune usernames)

### Pricing checks:
* [ ] `coin_id` represents the ID of the coin on coinpaprika.com
* [ ] all the coins are active on coinpaprika.com (please remove inactive ones)

### Join logic:
* [x] if joining to base table (i.e. ethereum transactions or traces), I looked to make it an inner join if possible

### Incremental logic:
* [x] I used is_incremental & not is_incremental jinja block filters on both base tables and decoded tables
  * [x] where block_time >= date_trunc("day", now() - interval '1 week')
* [x] if joining to base table (i.e. ethereum transactions or traces), I applied join condition where block_time >= date_trunc("day", now() - interval '1 week')
* [ ] if joining to prices view, I applied join condition where minute >= date_trunc("day", now() - interval '1 week')

### CODE REVIEW

File: models/synthetix/optimism/synthetix_optimism_perps_trades.sql [hunk 1]
```diff
@@ -0,0 +1,56 @@
+{{ config(
+    alias ='trades',
+    post_hook='{{ expose_spells(\'["optimism"]\',
+                                    "project",
+                                    "pika",
```
Comment (MSilb7) [inline models/synthetix/optimism/synthetix_optimism_perps_trades.sql:5]: replace `pika` with `synthetix`

File: models/synthetix/optimism/synthetix_optimism_perps_trades.sql [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+		,project
+		,version
+		,trader
+		,volume_raw
+		,tx_hash
+		,tx_from
+		,tx_to
+		,evt_index
+    FROM {{ ref('synthetix_optimism_perps_v1_trades')}}
+    UNION
```
Comment (MSilb7) [inline models/synthetix/optimism/synthetix_optimism_perps_trades.sql:34]: would `union all` make any impact on query runtime?

**Comment by MSilb7** [comment]
🙏 Left some small comments.

I see Kwenta and Decentrex mentioned in tags. Is there a way to extract which frontend was used (Polynomial also just made a frontend)? If so, does that make sense as part of this spell as well?

File: models/synthetix/optimism/synthetix_optimism_perps_trades.sql (hunk 1 above)
Comment (RplusT) [reply models/synthetix/optimism/synthetix_optimism_perps_trades.sql]: My bad. Good catch!

File: models/synthetix/optimism/synthetix_optimism_perps_trades.sql (hunk 2 above)
Comment (RplusT) [reply models/synthetix/optimism/synthetix_optimism_perps_trades.sql]: I'm not sure actually. Changed it to `UNION ALL` now

**Comment by RplusT** [comment]
@MSilb7 
> I see Kwenta and Decentrex mentioned in tags. Is there a way to extract which frontend was used (Polynomial also just made a frontend)? If so, does that make sense as part of this spell as well?

Yes this should be possible with the `trackingCode` parameter for the [market trading actions](https://docs.synthetix.io/integrations/perps-integration-guide/tracking-codes). However, the market modification calls that have tracking isn't required for trading (i.e frontends/protocols can choose to not be tracked) so I'll have to double check attributable volumes in that regard.

If ever, what would be a good way to label them in the abstraction? I was thinking `Synthetix_<frontend name>` under the `project` column instead of just plain `Synthetix`.

**Comment by MSilb7** [comment]
> @MSilb7
> 
> > I see Kwenta and Decentrex mentioned in tags. Is there a way to extract which frontend was used (Polynomial also just made a frontend)? If so, does that make sense as part of this spell as well?
> 
> Yes this should be possible with the `trackingCode` parameter for the [market trading actions](https://docs.synthetix.io/integrations/perps-integration-guide/tracking-codes). However, the market modification calls that have tracking isn't required for trading (i.e frontends/protocols can choose to not be tracked) so I'll have to double check attributable volumes in that regard.
> 
> If ever, what would be a good way to label them in the abstraction? I was thinking `Synthetix_<frontend name>` under the `project` column instead of just plain `Synthetix`.

I was thinking maybe there's a separate `frontend` column or something? Not all perps products would have a function like this, so maybe not relevant for perps.trades. Don't want to hold up this PR at all (and not sure what the error is).

Brainstorming:
An alternative could be a separate "frontend mapping" spell (longer-term). Since it seems like the separation of contracts and frontends is happening (i.e. Kwenta taking over Lyra's frontend, many wallets plugging in to 0x).

So maybe not relevant to live within Perps, but rather we should longer-term find a way like "how do we map transactions to their frontend?"

**Comment by RplusT** [comment]
@MSilb7 I like the idea of a `frontend` column. For the non-Synthetix projects, we can just put in the `project` name as the `frontend` as well.

I think we (along with the rest of the OP/Synthetix/Perps community) can figure out a longer term solution later on with the front-end mapping, especially once the community sees what we do here with the `frontend` column (maybe 😄)

**Comment by MSilb7** [comment]
> @MSilb7 I like the idea of a `frontend` column. For the non-Synthetix projects, we can just put in the `project` name as the `frontend` as well.
> 
> I think we (along with the rest of the OP/Synthetix/Perps community) can figure out a longer term solution later on with the front-end mapping, especially once the community sees what we do here with the `frontend` column (maybe 😄)

Yeah, i imagine a separate spell for "frontend mapping" would make sense, rather than refactoring every other spell haha. Like 'perps_frontends' dex_frontends' that all merge in to one big 'frontends' table which you could join too

**Comment by RplusT** [comment]
@MSilb7 exciting future work 😅. 

Also talk about timing, I just pushed the changes to add a `frontend` column across all the perpetuals trading projects when you replied.

**Comment by MSilb7** [comment]
> @MSilb7 exciting future work 😅
> 
> Also talk about timing, I just pushed the changes to add a `frontend` column across all the perpetuals trading projects when you replied.

Haha love it. Didn't want to put extra work on you, but super pumped to be able to also track frontend attribution!

**Review by Hosuke — APPROVED** [review]
LGTM.✅
This PR adds `frontend` to all perpetual_trades models, and a new model `synthetix_optimism_perps_v2_trades` to synthetix_perps_trades.
Thank you @RplusT

File: seeds/perpetual/trades/perpetual_trades_seed.csv [hunk 3]
```diff
@@ -20,3 +20,5 @@ avalanche_c,2023-01-27,,,0x3cf47469298b1630c192b2c8d3ffa087abcac8b2,close-short,
 avalanche_c,2023-01-27,,,0x3cf47469298b1630c192b2c8d3ffa087abcac8b2,open-long,emdx,1,0xd456e4ab2ddfc1e877424869ed91f590c58779092d4514263b039e233c8d639a
 avalanche_c,2023-01-26,AVAX,AVAX,0x4e3535964cb5612a466d8bb25362d485452efcef,open-short,hubble_exchange,1,0x3f187beac116995fa84a8185fd5c220c9c0dd6b1b7af737ab83065ae3da81329
 avalanche_c,2023-01-18,AVAX,AVAX,0x4e3535964cb5612a466d8bb25362d485452efcef,close-short,hubble_exchange,1,0x984182f9e697acc213bcaf6c87858ebe7e0cbeb99019a73b5100e0ec837e7e96
+optimism,2023-02-26,DYDX,sDYDXPERP,0x139f94e4f0e1101c1464a321cba815c34d58b5d9,long,Synthetix,2,0xa9900b71d779aec64fc08fb96a770677e4b6402cc67ec5ab96e08baf25731b93
+optimism,2023-02-26,ETH,sETHPERP,0x2b3bb4c683bfc5239b029131eef3b1d214478d93,short,Synthetix,2,0x67d7039204428c2c66e058c81fe6632261df3aba0ba04b82442ff2a62a6abbd9
```
Comment (Hosuke) [inline seeds/perpetual/trades/perpetual_trades_seed.csv:24]: Fix the block_date in the last commit, as 2023 is mistakenly typed as 2022.

**Comment by MSilb7** [comment]
hey, wondering if this is good to merge! Would love to have our Perps charts get updated.

**Review by jeff-dude — CHANGES_REQUESTED** [review]
apologies to continue to push the scope of this PR, but since it's already editing existing spells within `perpetual.trades` -- i wonder if we can use this to further enhance. with the new spells being added, there seems to be some inconsistency in naming standards. i'd love to see this spell continue to grow similar to `dex.trades`.

some thoughts:
- since some protocols/projects have both perpetual and dex trades (from what i can see in gmx), i wonder if we revisit naming standard for consistency
- can we ensure all file names in the repo are: `<project_name>_<version>_<blockchain>_perpetual_trades`?
- schema is: `<project_name>_<version>_<blockchain>` if version, applicable, otherwise without?
- all base incremental tables / rolled up protocols across versions/blockchains views use alias: `perpetual_trades`, then only the final spell is just `perpetual.trades`?

also for a consistency/readability enhancement, similar to dex trades, can we use the for loop jinja syntax on all rollup views in this spell too? you can find the example within `dex.trades` file

**Comment by RplusT** [comment]
@jeff-dude great points regarding naming consistency. definitely agree; don't wanna have this grow to be unwieldy. I made the naming changes. Let me know what you think. :)

**Comment by jeff-dude** [comment]
all old tables/views which were renamed here should be removed from the db

**Pull request merged**